src/main/java/com/libertymutualgroup/herman/aws/ecs/loadbalancing/EcsLoadBalancerHandler.java

```diff
@@ -84,37 +83,34 @@ public LoadBalancer createLoadBalancer(EcsClusterMetadata clusterMetadata, EcsPu
         }
 
         String urlSuffix = definition.getService().getUrlSuffix();
-        DeriveCertResult deriveCertResult = certHandler.deriveCert(protocol, urlSuffix, urlPrefix);
+        SSLCertificate sslCertificate = certHandler.deriveCert(protocol, urlSuffix, urlPrefix);
 
         ContainerDefinition webContainer = portHandler.findContainerWithExposedPort(definition, false);
         Integer randomPort = webContainer.getPortMappings().get(0).getHostPort();
         Integer containerPort = webContainer.getPortMappings().get(0).getContainerPort();
         String containerName = webContainer.getName();
 
-        boolean isInternetFacingUrlScheme = certHandler.isInternetFacingUrlScheme(deriveCertResult.getSslCertificate(),
+        boolean isInternetFacingUrlScheme = certHandler.isInternetFacingUrlScheme(sslCertificate,
             definition.getService().getUrlSchemeOverride());
-        boolean isUsingInternalSubnets = true;
         String elbScheme;
         List<String> elbSubnets;
         if (isInternetFacingUrlScheme || "internet-facing".equals(definition.getService().getElbSchemeOverride())) {
             elbScheme = "internet-facing";
-            isUsingInternalSubnets = false;
             elbSubnets = clusterMetadata.getPublicSubnets();
         } else {
             elbScheme = "internal";
             elbSubnets = clusterMetadata.getElbSubnets();
         }
 
-        // Handle akamai
         List<String> elbSecurityGroups = new ArrayList<>();
-        if (isInternetFacingUrlScheme && HTTPS.equals(protocol)) {
-            elbSecurityGroups.addAll(clusterMetadata.getAkamaiSecurityGroup());
+        if ("internet-facing".equals(elbScheme) && HTTPS.equals(protocol)) {
```

Comment:
Minor but I wonder if checking for "internet-facing" should go into a helper method?

Reply:
Good thought, I'll take a look at refactoring this later